python/ShardArchiver.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+#    Copyright (c) 2018 Ripple Labs Inc.
+#
+#    Permission to use, copy, modify, and/or distribute this software for any
+#    purpose  with  or without fee is hereby granted, provided that the above
+#    copyright notice and this permission notice appear in all copies.
+#
+#    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+#    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+#    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+# Creates lz4 compressed tar archives from complete
+# shards and optinally SCP transfers to a destination
+
+import lz4.frame
+import os
+import subprocess
+import sys
+import tarfile
+
+from os import listdir, remove
+from os.path import basename, isdir, isfile, join
+
+
+def validate_args():
+    '''Validate command line arguments'''
+
+    usage = ("usage: shard_archiver <shards_directory> <output_directory>"
+             " [<indenty_file> <username> <host> <host_directory>]\n"
+             "example: shard_archiver.py /db/shards . /.ssh/id_dsa "
+             "username domain.com \\\\home\\\\archives\n")
+
+    arg_count = len(sys.argv) - 1
+    if arg_count != 2 and arg_count != 6:
+        print('Invalid number of arguments.\n')
+        print(usage)
+        sys.exit(1)
+
+    # Sanity check the shards DB path
+    if not isdir(sys.argv[1]):
+        print('Invalid shards directory.\n')
+        print(usage)
+        sys.exit(1)
+
+    # Sanity check the output path
+    if not isdir(sys.argv[2]):
+        print('Invalid output directory.\n')
+        print(usage)
+        sys.exit(1)
+
+    if arg_count > 3:
+        # Sanity check the identity file
+        if not isfile(sys.argv[3]):
+            print('Invalid identity file.\n')
+            print(usage)
+            sys.exit(1)
+
+
+def read_chunk(file_object):
+    while True:
+        data = file_object.read(16384)
+        if not data:
+            break
+        yield data
+
+
+def create_lz4(source_file, output_path):
+    src = open(source_file, mode='rb')
+    with lz4.frame.open(output_path,
+                        mode='wb',
+                        block_size=lz4.frame.BLOCKSIZE_MAX1MB,
+                        compression_level=lz4.frame.COMPRESSIONLEVEL_MAX,
+                        content_checksum=True) as f:
+        for piece in read_chunk(src):
+            f.write(piece)
+
+
+def process(args):
```

Comment:
I'd give the arguments individual names rather than just passing in `args`. If the argument is optional (as with indenty_file, username, host, and host_directory), just default them to `None`.